pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java

```diff
@@ -148,10 +163,9 @@ protected String getQuery(SlingHttpServletRequest request) {
         productCondition = "AND (" + StringUtils.join(conditions, " OR ") + ") ";
 
         StringBuilder query = new StringBuilder()
-                .append("SELECT pv.* from [nt:base] AS pv ")
-                .append("INNER JOIN [nt:base] AS product ON ISDESCENDANTNODE(pv, product) ")
-                .append("WHERE pv.[jcr:primaryType] = 'pant:productVersion' ")
-                .append("AND product.[jcr:primaryType] = 'pant:product' ")
+                .append("SELECT pv.* from [pant:productVersion] AS pv ")
+                .append("INNER JOIN [pant:product] AS product ON ISDESCENDANTNODE(pv, product) ")
```

Comment:
We still have this inner join - is it worth eliminating? We could probably easily use deep attribute inspection here.

Reply:
This one is trickier to turn into XPATH, as we are actually filtering on a parent node's properties from the child. A single join doesn't seem to be as costly also. In addition, this one is only being executed when the product filters are used, so any additional cost is limited to those scenarios.

Reply:
Ooof. Good point. Alright, we can save this one for later/never.